### PR TITLE
feat: big-endian support

### PR DESCRIPTION
**Pull request by paul-soporan**
**What's the problem this PR addresses?**

This PR adds big-endian support to Yarn.

Closes https://github.com/yarnpkg/berry/issues/2745.

**How did you fix it?**

Changes required to get Yarn to work on big-endian systems:

- [X] Recompiling libzip with `-s SUPPORT_BIG_ENDIAN=1`
- [X] Lazy-import ink everywhere and throw an error stating that interactive commands aren't supported on BE systems (~~Recompiling `yoga-layout-prebuilt` with `-s SUPPORT_BIG_ENDIAN=1`~~ Recompiling `yoga-layout-prebuilt` requires recompiling `yoga-layout` whose last release was 2 years ago. Unfortunately, compiling yoga is a nightmare because of changes in newer versions of node, clang, and possibly emscripten.)

**Checklist**

- [X] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [X] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by namrata-ibm** [comment]
@paul-soporan Having this merged would be really great! Could you please help?

**Comment by namrata-ibm** [comment]
@paul-soporan @arcanis gentle reminder!

**Comment by chrandgull** [comment]
> @paul-soporan @arcanis gentle reminder!

A simple 'thank you' from the mainframe community for developing this patch. :) 

I was able to compile and build Grafana v8.5.0 with this. Great work!

**Comment by paul-soporan** [comment]
Everything on my side should be finished other than the occasional rebase, just waiting on a review now (CC @arcanis)

File: .yarn/versions/11abead6.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  "@yarnpkg/cli": minor
+  "@yarnpkg/core": minor
+  "@yarnpkg/fslib": minor
+  "@yarnpkg/libui": minor
+  "@yarnpkg/libzip": minor
+  "@yarnpkg/plugin-interactive-tools": minor
+  "@yarnpkg/plugin-pnp": minor
+  "@yarnpkg/plugin-version": minor
+  "@yarnpkg/pnp": minor
+  vscode-zipfs: minor
```
Comment (merceyz) [inline .yarn/versions/11abead6.yml:11]: @paul-soporan Could this be marked as a `patch` instead so I can backport it?
I see the reason for `minor` (adding big-endian support) but one could also look at it as a `patch` (fixing big-endian support).

**Review by merceyz — DISMISSED** [review]

**Review by arcanis — APPROVED** [review]

**Pull request merged**